lxd/container_exec.go

```diff
@@ -391,52 +391,46 @@ func containerExecPost(d *Daemon, r *http.Request) response.Response {
 	}
 
 	// Process environment.
-	env := map[string]string{}
-	for k, v := range inst.ExpandedConfig() {
-		if strings.HasPrefix(k, "environment.") {
-			env[strings.TrimPrefix(k, "environment.")] = v
-		}
+	if post.Environment == nil {
+		post.Environment = map[string]string{}
 	}
 
-	if post.Environment != nil {
-		for k, v := range post.Environment {
-			env[k] = v
+	for k, v := range inst.ExpandedConfig() {
+		if strings.HasPrefix(k, "environment.") {
+			post.Environment[strings.TrimPrefix(k, "environment.")] = v
```

Comment:
This looks wrong.

Prior to this we were taking the environement set in the instance config, then override that with what's set in req.Environment, then set additional keys.

With your change, you're now overriding req.Environment with what's in the instance config, so wrong way around.

Reply:
@stgraber thanks Ive pushed a fixed for that.